plugins/org.eclipse.fordiac.ide.export/src/org/eclipse/fordiac/ide/export/Exporter.java

```
@@ -16,31 +16,45 @@
 import java.nio.file.Path;
 import java.text.MessageFormat;
 import java.util.List;
+import java.util.Optional;
 
 import org.eclipse.core.resources.IFile;
-import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IConfigurationElement;
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.OperationCanceledException;
 import org.eclipse.emf.ecore.EObject;
+import org.eclipse.fordiac.ide.export.utils.ExportFilterUtil;
 import org.eclipse.fordiac.ide.model.libraryElement.INamedElement;
 import org.eclipse.fordiac.ide.model.typelibrary.CMakeListsMarker;
+import org.eclipse.fordiac.ide.ui.FordiacLogHelper;
 import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.MessageBox;
+
+public class Exporter {
 
-public abstract class AbstractExporter {
 	private final String outputDirectory;
 	private final IConfigurationElement filterConfig;
 	private boolean overwriteWithoutWarning;
 	private boolean enableCMakeLists;
 	private final IExportFilter filter;
 
+	public Exporter(final IConfigurationElement filterConfig, final String outputDirectory,
+			final boolean overwriteWithoutWarning, final boolean enableCMakeLists) {
+		this.filterConfig = filterConfig;
+		this.outputDirectory = outputDirectory;
+		this.overwriteWithoutWarning = overwriteWithoutWarning;
+		this.enableCMakeLists = enableCMakeLists;
+		this.filter = ExportFilterUtil.createExportFilter(Optional.of(filterConfig));
+
+	}
+
 	public void exportElements(final IProgressMonitor monitor, final List<IFile> exportees) {
 		monitor.beginTask(MessageFormat.format(Messages.FordiacExporter_ExportingSelectedTypesUsingExporter,
 				filterConfig.getAttribute("name")), exportees.size() + 1); //$NON-NLS-1$
 
 		if (null != filter) {
 			for (final IFile file : exportees) {
-				if (!exportIsCanceled(monitor)) {
+				if (!monitor.isCanceled()) {
```

I would suggest also throwing an `OperationCanceledException` here, same as below, instead of continuing with the loop.